### ISSUE TITLE
[Error] distrobox-rm fails even if container is stopped

### ISSUE THREAD

**Issue by paidhi** [89luca89/distrobox#1335]
I cannot remove a distrobox container without using `--force`.

It looks to me like `distrobox-rm` is starting the container (running `distrobox-enter`) to find out about exported apps.
But it does not stop the container before trying to remove it.

I have attached the output of `distrobox-rm --verbose` for details.

Versions:

```text
# Host distribution:
> grep PRETTY_NAME /etc/os-release
PRETTY_NAME="Fedora Linux 39 (Thirty Nine)"

> rpm -q podman
podman-4.9.4-1.fc39.x86_64

> rpm -q distrobox
distrobox-1.7.1-2.fc39.noarch

> cat ~/.config/distrobox/distrobox.conf
container_generate_entry=0
container_image_default="registry.fedoraproject.org/fedora-toolbox:39"
container_name_default="fedora"
non_interactive=1
skip_workdir=1
```

How to reproduce:

```text
distrobox create --name fedora --image registry.fedoraproject.org/fedora-toolbox:39

distrobox enter fedora

# Exit distrobox container
fedora> exit

> distrobox list
ID           | NAME                 | STATUS             | IMAGE
1acf43d517ae | fedora               | Up About a minute  | registry.fedoraproject.org/fedora-toolbox:39

> distrobox stop fedora
fedora

> podman container ls -a
CONTAINER ID  IMAGE                                         COMMAND               CREATED        STATUS                       PORTS       NAMES
1acf43d517ae  registry.fedoraproject.org/fedora-toolbox:39  --verbose --name ...  4 minutes ago  Exited (143) 26 seconds ago              fedora
> distrobox list
ID           | NAME                 | STATUS             | IMAGE
1acf43d517ae | fedora               | Exited (143) 35 seconds ago | registry.fedoraproject.org/fedora-toolbox:39

> distrobox rm fedora
Removing exported binaries...
Removing container...
Error: cannot remove container 1acf43d517ae7c2cf8e4fa4d33542ad7c3e94c832c1ea37027a02cb6fbd4710c as it is running - running or paused containers cannot be removed without force: container state improper

> podman container ls
CONTAINER ID  IMAGE                                         COMMAND               CREATED        STATUS         PORTS       NAMES
1acf43d517ae  registry.fedoraproject.org/fedora-toolbox:39  --verbose --name ...  5 minutes ago  Up 32 seconds              fedora
> distrobox list
ID           | NAME                 | STATUS             | IMAGE
1acf43d517ae | fedora               | Up 35 seconds      | registry.fedoraproject.org/fedora-toolbox:39
```

To successfully remove the distrobox I have to force it:

```text
> distrobox stop fedora
fedora

> distrobox list
ID           | NAME                 | STATUS             | IMAGE
1acf43d517ae | fedora               | Exited (143) 5 seconds ago | registry.fedoraproject.org/fedora-toolbox:39

> distrobox rm --force fedora
Removing exported binaries...
Removing container...
fedora
```

[distrobox-rm_verbose.log](https://github.com/89luca89/distrobox/files/14932369/distrobox-rm_verbose.log)

**Comment by YatesRocks** [comment]
I'm getting this same issue on openSUSE Tumbleweed w/version 1.7.1.0

**Comment by 89luca89** [comment]
Fixed by ac34f1050bc14860b415541657c15991a019698f

**Comment by paidhi** [comment]
Thanks!
I can confirm that the issue does not occur anymore with version 1.7.2.0.
